### PR TITLE
Adding azure FS improvements

### PR DESCRIPTION
**Pull request by jenniferdai**
* Supporting copy on ADL
* Ignoring old file during fs methods if it exists
* Validated through local testing on ADL

### CODE REVIEW

File: pinot-hadoop-filesystem/src/main/java/com/linkedin/pinot/filesystem/AzurePinotFS.java [hunk 1]
```diff
@@ -88,8 +91,15 @@ public boolean move(URI srcUri, URI dstUri) throws IOException {
 
   @Override
   public boolean copy(URI srcUri, URI dstUri) throws IOException {
-    // TODO: Add this method. Not needed for now because file will be in final location
-    throw new UnsupportedOperationException("Cannot copy from AzureFS to AzureFS");
+    if (exists(dstUri)) {
```
Comment (mayankshriv) [inline pinot-hadoop-filesystem/src/main/java/com/linkedin/pinot/filesystem/AzurePinotFS.java]: Should we have overwrite as parameter passed? So we don't always overwrite.

File: pinot-hadoop-filesystem/src/main/java/com/linkedin/pinot/filesystem/AzurePinotFS.java [hunk 2]
```diff
@@ -117,6 +127,13 @@ public long length(URI fileUri) throws IOException {
 
   @Override
   public void copyToLocalFile(URI srcUri, File dstFile) throws Exception {
+    if (dstFile.exists()) {
```
Comment (mayankshriv) [inline pinot-hadoop-filesystem/src/main/java/com/linkedin/pinot/filesystem/AzurePinotFS.java]: Same here.

File: pinot-hadoop-filesystem/src/main/java/com/linkedin/pinot/filesystem/AzurePinotFS.java (hunk 1 above)
Comment (jenniferdai) [reply pinot-hadoop-filesystem/src/main/java/com/linkedin/pinot/filesystem/AzurePinotFS.java]: I think we don't have that use case in Pinot and I don't anticipate us needing that as these are used for segment upload, refreshing segments on server side, etc.

Comment (jenniferdai) [reply pinot-hadoop-filesystem/src/main/java/com/linkedin/pinot/filesystem/AzurePinotFS.java]: When we designed, other filesystems default overwrite, so I wanted to keep it consistent.

**Pull request merged**